src/tags/include_tag.rs

```diff
@@ -29,25 +30,28 @@ impl Renderable for Include {
 fn parse_partial(name: &str, options: &LiquidOptions) -> Result<Template> {
     let content = options.include_source.include(name)?;
 
-    let tokens = tokenize(&content)?;
-    parse(&tokens, options).map(Template::new)
+    parse(&content, options).map(Template::new)
 }
 
 pub fn include_tag(
     _tag_name: &str,
-    arguments: &[Token],
+    arguments: &mut Iterator<Item=TagToken>,
     options: &LiquidOptions,
 ) -> Result<Box<Renderable>> {
-    let mut args = arguments.iter();
-
-    let name = match args.next() {
-        Some(&Token::StringLiteral(ref name)) => name,
-        Some(&Token::Identifier(ref s)) => s,
-        arg => return Err(unexpected_token_error("string", arg)),
+    let name = arguments.next().unwrap_or_else(|| panic!("Errors not implemented. Token expected."));
+    // TODO: make `name` a &str instead
+    let name = if let Ok(name) = name.expect_identifier() {
+        name.to_string()
+    } else if let Ok(name) = name.expect_literal() {
+        // This will allow non string literals such as 0 to be parsed as such.
+        // Is this ok or should more specific functions be created?
```

Comment:
Not dug into this yet but don't we need a way to turn a literal into a `Scalar` which requires knowing if it is a string literal or not?  Can't we leverage that?

Reply:
The thing is currently there is a function `parse_literal` that automatically works for any literal, `String`, `Boolean`, `Float`, `Integer`, turning them into the respective `Scalar`.

The function `expect_literal` allows any literal to be parsed, returning a `Value`. I think this is fine for most use cases.
Now for cases like this one, I'm unsure what the behavior of liquid is supposed to be. If its fine for it to accept other literals as well, this is the way to go. Otherwise, more functions must be created, to expect specific types of Literals.